### DISCUSSION TITLE
detect ubuntu apt packages

### DISCUSSION THREAD

**Discussion by SBatais** [renovatebot/renovate#25028] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.5.3

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I would like that renovate detects dependencies for ubuntu apt packages that I install inside a Dockerfile.
Renovate is configured with the default settings, and I would like to know if renovate is able by default to detect this type of dependencies or if I have to configure something (new datasource for example)
Here's my Dockerfile in which I install the apt packages python3 and python3-pip

```
FROM ubuntu:focal
RUN apt-get update && \
  apt-get install -y --no-install-recommends \
  python3=3.8.2-0ubuntu2 \
  python3-pip=20.0.2-5ubuntu1.9
```
Thanks in advance for your help

### What you tried so far.

I didn't tried anything

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by SBatais** [comment]
I think that I have to configure something with the datasource repology, but I don't know how

**Reply by HonkingGoose** [reply]
## Renovate `repology` datasoure docs

Did you read the [Renovate `repology` datasource docs](https://docs.renovatebot.com/modules/datasource/repology/)?

## Try `loose` or `debian` versioning first

> **Tip**
> We recommend you try `loose` or `debian` versioning for distribution packages first. This is because the version number usually doesn't match Renovate's default `semver-coerced` specification.

- [Renovate docs, `debian` versioning](https://docs.renovatebot.com/modules/versioning/#debian-versioning)
- [Renovate docs, `loose` versioning](https://docs.renovatebot.com/modules/versioning/#loose-versioning)

**Comment by viceice** [comment, marked as answer]
that won't work out of the box. there are so many ways to write that code. to complex to make that build-in.

you need to use a regex manager to extract your deps manually

**Comment by MPV** [comment]
If you're up for trying the `deb` datasource instead of `repology`, I have a working example here:
- https://github.com/MPV/renovate-ubuntu-locales

**Reply by jamietanna** [reply]
Not OP, but using your `deb` datasource worked very nicely with an Ubuntu Focal (20.04) package I was looking at updating :rocket:
